src/cmd/ksh93/tests/arith.sh

```diff
@@ -761,4 +761,10 @@ x=0x1.0000000000000000000000000000p+6
 v=$(printf $'%.28a\n' 64)
 [[ $v == "$x" ]] || err_exit "'printf %.28a 64' failed -- expected '$x', got '$v'"
 
+# ======
+# Redirections with ((...)) should not cause a syntax error
+"$SHELL" 2>/dev/null -c '(($(echo 1+1 | tee /dev/fd/3))) >/dev/null 3>&1'
```

Comment:
`/dev/fd/` is not portable. I also don't understand what that whole command substitution is for. A simple way to test for this syntax error would be:
```sh
(eval '((1)) >/dev/null') 2>/dev/null || err_exit 'redirections with ((...)) yield a syntax error'
```
Using `eval` avoids triggering a syntax error prematurely on a ksh with the bug.